analytics/transform.go

```diff
@@ -197,16 +195,101 @@ func (event ParsedEvent) GetValue(field string) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	if field == "contexts" || field == "derived_contexts" || field == "unstruct_event" {
-		// TODO: DRY HERE TOO?
-		output := make(map[string]interface{})
-		for _, pair := range kvPairs {
-			output[pair.Key] = pair.Value
-		}
-		return output, nil
+
+	return kvPairs, nil
+}
+
+// GetValue returns the value for a provided atomic field, without processing the rest of the event.
+// For unstruct_event, it returns a map of only the data for the unstruct event.
+func (event ParsedEvent) GetValue(field string) (interface{}, error) {
+	kvPairs, err := event.getParsedValue(field)
+	if err != nil {
+		return nil, err
 	}
 	return kvPairs[0].Value, nil
+}
+
+// GetUnstructEventValue returns the value for a provided atomic field inside an event's unstruct_event field
+//
+func (event ParsedEvent) GetUnstructEventValue(field string, path ...interface{}) (interface{}, error) {
+	if field != "unstruct_event" {
+		return nil, errors.New(`Invalid path, not inside unstruct_event`)
+	}
+
+	kvPairs, err := event.getParsedValue(field)
+	if err != nil {
+		return nil, err
+	}
+
+	// extract the key/value pairs of the event field into a map
+	eventMap := make(map[string]interface{})
+	for _, pair := range kvPairs {
+		eventMap[pair.Key] = pair.Value
+	}
 
+	j, err := json.Marshal(eventMap)
+	if err != nil {
+		return nil, err
+	}
+
+	b := make([]interface{}, len(path))
+	for idx := range path {
+		b[idx] = path[idx]
+	}
+
+	el := json.Get(j, b...)
+	return el.GetInterface(), nil
+}
+
+// GetContextValue returns the value for a provided atomic field inside an event's contexts or derived_contexts
+func (event ParsedEvent) GetContextValue(field string, path ...interface{}) (interface{}, error) {
+	if field != "contexts" && field != "derived_contexts" {
+		return nil, errors.New(`Invalid path, not a context`)
+	}
```

Comment:
This makes more sense here, however I wonder if this is what people expect? 

If we're being true to the event format, then the difference between `contexts` and `derived_contexts` is important (the latter are contexts added by enrichment, rather than trackers). However, in reality, would a user actually care if we think about where they are using this SDK, after the enriched stream?

I have a feeling if we could hide this implementation detail on the API to this function, then that might lead to a better experience for the users. Although it'll be a little messier inside this function. We perhaps have two options: 
1. Search `contexts` first, then `derived_contexts`, merge the results and return them.
2. Merge contexts and derived_contexts then find the searched for context.

Either way, worth bearing in mind that a "search" context could appear in either, neither or both of `contexts` and `derived_contexts` and it could appear multiple times across them.